src/postings/postings.service.ts

```diff
@@ -0,0 +1,109 @@
+import { PostingResponseDto } from './dto/posting-response.dto';
+import { PostingQueryDto } from './dto/posting-query.dto';
+
+@Injectable()
+export class PostingsService {
+  constructor(@InjectRepository(Posting) private readonly postingRepository: Repository<Posting>) {}
+
+  /**
+   * 게시물의 목록을 조회합니다. content는 20자 까지만 조회됩니다.
+   * @param queries
+   * @returns 게시물 목록. 없는 경우 빈 배열
+   */
+  async findAll(hashtag: string, queries: PostingQueryDto): Promise<PostingResponseDto[]> {
+    const rawPostings = await this.getRawPostings(hashtag, queries);
+
+    // hashtag depth 정리 & content 글자 수 제한 적용해 반환
+    return rawPostings.map((rawPosting) => {
+      const posting = this.getPostingObjWithHashtags(rawPosting);
+      posting.content = posting.content.substring(0, 20);
+
+      return posting;
+    });
+  }
+
+  /**
+   * 게시물을 상세 조회합니다. 조회수가 1씩 증가합니다.
+   * @param id 게시물 PK(uuid)
+   * @returns id에 해당하는 게시물
+   */
+  async findOne(id: string): Promise<PostingResponseDto> {
+    const posting = await this.postingRepository.findOne({
+      where: { id },
+      relations: ['postingHashtags', 'postingHashtags.hashtag'],
+    });
+
+    if (posting === null) {
+      throw new NotFoundException('posting not found');
+    }
+    this.postingRepository.update(id, { viewCount: posting.viewCount + 1 });
+
+    return this.getPostingObjWithHashtags(posting);
+  }
+
+  // 입력된 쿼리 파라미터를 바탕으로 DB 쿼리를 빌드해 게시물 목록을 가져옵니다.
+  private async getRawPostings(hashtag: string, queries: PostingQueryDto): Promise<Posting[]> {
+    const { type, orderBy, sortOrder, searchBy, search, pageCount, page } = queries;
+
+    const queryBuilder = this.postingRepository.createQueryBuilder('posting');
+
+    // 1건의 hashtag와 정확히 일치하는 값 검색
+    queryBuilder
+      .innerJoinAndSelect('posting.postingHashtags', 'postingHashtag')
+      .innerJoinAndSelect('postingHashtag.hashtag', 'hashtag')
+      .where('hashtag.name = :hashtag', { hashtag });
+
+    // 게시물의 type 별로 조회, 미입력 시 모든 type
+    if (type) {
+      queryBuilder.andWhere('posting.type = :type', { type });
+    }
+
+    // search 입력된 경우 searchBy에 따라 검색
+    if (search) {
+      if (searchBy.split(',').length === 2) {
+        queryBuilder.andWhere('posting.title LIKE :search OR posting.content LIKE :search', { search: `%${search}%` });
+      } else {
+        queryBuilder.andWhere(`posting.${searchBy} LIKE :search`, { search: `%${search}%` });
+      }
+    }
+
+    // 입력된 값대로 정렬해 페이지네이션
+    const rawPostings = await queryBuilder
+      .orderBy(`posting.${orderBy}`, sortOrder)
+      .skip(page * pageCount)
+      .take(pageCount)
+      .getMany();
+
+    return rawPostings;
+  }
+
+  // 조인으로 인한 hashtag의 depth를 정리한 새 객체를 반환합니다.
+  private getPostingObjWithHashtags(postingObj: Posting): PostingResponseDto {
+    const hashtags: string[] = [];
+    if (postingObj.postingHashtags.length) {
+      postingObj.postingHashtags.forEach((v) => {
+        hashtags.push(v.hashtag.name);
+      });
+    }
```

Comment:
```suggestion
const hashtags: string[] = postingObj.postingHashtags.map(postingHashtag) => postingHashtag.hashtag.name);
```
참고용 suggestion : `map`을 사용하면 return 으로 바로 새 배열을 생성할 수 있습니다. 앗 위에 `findAll` 에서는 `map`을 사용하셨는데 여기서는  `forEach` 사용하신 이유가 혹시 있을까요?
그리고 length 체크 로직 사용한 이유가 있으신가요? length가 0이면 `forEach()` 실행이 안되는 걸로 알아서 의미 없는 부분같은데 특별한 이유가 있으신지 궁금합니다.

Reply:
말씀 주신 두 부분이 같이 가는거라, length 체크를 위해 `map` 대신 `forEach`를 사용했습니다.
그러나 특별한 이유는 없었습니다... `v.hashtag.name` 할 때 객체에 없는 키값 읽는 에러 피하려고 썼던건데, 0이면 메서드 내에서도 그냥 넘어간다는걸 이땐 잊고 있었네요;; 감사합니다!!